extensions/github1s/src/source-control/changes.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file source control change file list
+ * @author netcon
+ */
+
+import * as vscode from 'vscode';
+import repository, { FileChangeType } from '@/repository';
+import router from '@/router';
+import { basename } from '@/helpers/util';
+import { GitHub1sFileSystemProvider } from '@/providers/fileSystemProvider';
+import { PageType } from '@/router/types';
+
+interface ChangedFile {
+	previousFileUri: vscode.Uri;
+	currentFileUri: vscode.Uri;
+	status: FileChangeType;
+}
+
+const getChangedFiles = async (): Promise<ChangedFile[]> => {
+	const routerState = await router.getState();
+	// current workspace root directory uri, the `ref` in uri authority should be `pull.head.sha`
+	let currentRootUri = vscode.Uri.parse('').with({
+		scheme: GitHub1sFileSystemProvider.scheme,
+		authority: await router.getAuthority(),
+		path: '/',
+	});
+
+	// github pull page
+	if (routerState.pageType === PageType.PULL) {
+		// record that current editor is opened by pull
+		currentRootUri = currentRootUri.with({
+			query: `pull=${routerState.pullNumber}`,
+		});
+		const { owner, repo } = await router.getState();
+		const baseRef = (await repository.getPull(routerState.pullNumber)).base.sha;
+		const baseRootUri = currentRootUri.with({
+			authority: `${owner}+${repo}+${baseRef}`,
+		});
+
+		const pullFiles = await repository.getPullFiles(routerState.pullNumber);
+
+		return pullFiles.map((pullFile) => {
+			// the `previous_filename` field only exists in `RENAMED` file,
+			// fallback to `filename` otherwise
+			const previousFilePath = pullFile.previous_filename || pullFile.filename;
+			const currentFilePath = pullFile.filename;
+
+			return {
+				previousFileUri: vscode.Uri.joinPath(baseRootUri, previousFilePath),
```

Comment:
It fails to handle rename file:

https://github1s-git-feature-github-pull-vscode-github1s.vercel.app/xcv58/github1s/pull/16

![image](https://user-images.githubusercontent.com/503123/111050957-90435700-8404-11eb-841f-8ee6bbafdbe0.png)

Reply:
It might be caused by a bug on the GitHub side.
Let me reproduce.

Reply:
Confirmed it's a bug on the GitHub side https://github1s-git-feature-github-pull-vscode-github1s.vercel.app/xcv58/github1s/pull/17